app/src/main/java/ch/blinkenlights/android/vanilla/LibraryActivity.java

```diff
@@ -419,6 +432,20 @@ private void pickSongs(Intent intent, final int action)
 		query.mode = modeForAction[effectiveAction];
 		PlaybackService.get(this).addSongs(query);
 
+		//Delay this a short time, because sometimes the playlist is not ready and the queue is empty so it cannot set the track properly
```

Comment:
No: if anything, that should happen in a callback.

Sleeping a hardcoded amount of time will always be a source of bugs (and a horrible hack at best).